src/java/htsjdk/samtools/SamFlagField.java

```diff
@@ -0,0 +1,130 @@
+package htsjdk.samtools;
+
+/**
+ * This determines how flag fields are represented in the SAM file.
+ *
+ * In a string FLAG, each character represents one bit with
+ * p=0x1 (paired), P=0x2 (properly paired), u=0x4 (unmapped),
+ * U=0x8 (mate unmapped), r=0x10 (reverse), R=0x20 (mate reverse)
+ * 1=0x40 (first), 2=0x80 (second), s=0x100 (not primary), 
+ * x=0x200 (failure), d=0x400 (duplicate), and S=0x800 (secondary).
+ * This was inspired by 'samtools view -X'.
+ *
+ * We also output a character when the following bits *are not* set:
+ * m=0x4 (mapped), M=0x8 (mate mapped), f=0x10 (forward), F=0x20 
+ * (mate forward).
+ * 
+ * @author nhomer
+ */
+public enum SamFlagField {
+    NONE {
+        @Override
+        public String format(final int flag) {
+            throw new SAMException("NONE not allowed for the SamFlagField when writing the SAM flag field.");
+        }
+        @Override
+        public int parse(final String flag) {
+            throw new SAMException("NONE not allowed for the SamFlagField when reading the SAM flag field.");
+        } 
+    },
+    DEFAULT {
+        @Override
+        public String format(final int flag) {
+            return Integer.toString(flag);
+        }
+        /** Throws NumberFormatException if it can't parse the flag **/
+        @Override
+        public int parse(final String flag) { return Integer.parseInt(flag); }
+    },
+    HEXADECIMAL {
+        @Override
+        public String format(final int flag) {
+            return String.format("%#x", flag);
+        }
+        @Override
+        public int parse(final String flag) {
+            if (flag.startsWith("0x")) return Integer.valueOf(flag.substring(2), 16);
+            else return Integer.valueOf(flag, 16); // this really should not be supported, it should always have 0x..
+        }
+    },
+    OCTAL {
```

Comment:
I will leave that as a future improvement, as the functionality I am trying to re-create was in `samtools` originally, and did not have binary.  Also, this may confuse the auto-format recognition code depending on the binary output format.